src/time/src/helpers.rs

```diff
@@ -0,0 +1,41 @@
+use constants::*;
+use platform::types::*;
+
+// compute year, month, day & day of year
+// for description of this algorithm see
+// http://howardhinnant.github.io/date_algorithms.html#civil_from_days
+#[inline(always)]
+pub(crate) fn civil_from_days(days: c_long) -> (c_int, c_int, c_int, c_int) {
+    let (era, year): (c_int, c_int);
+    let (erayear, mut yearday, mut month, day): (c_int, c_int, c_int, c_int);
+    let eraday: c_ulong;
+
+    era = (match days >= 0 {
```

Comment:
Instead of these `match true/false` statements `if else` statements would look slightly better. Clippy also [warns about this](https://rust-lang-nursery.github.io/rust-clippy/v0.0.190/index.html#match_bool).